### PR TITLE
Removed iteratees and many deprecated methods

### PR DESCRIPTION
**Pull request by jroper**
* Iteratees now live at https://github.com/playframework/play-iteratees
* Iteratees reactive streams implemntation was also moved there
* Removed all deprecated methods/classes that use iteratees
* Removed all deprecated Java methods of handling streams (Chunks, LegacyWebSocket)
* Removed F.Promise
* Removed GlobalSettings
* Removed most deprecated methods from play.Play/play.api.Play (only left methods for accessing the application).

One thing that this required was updating the WebSockets documentation - that part of this PR would be useful to backport to 2.5.

Since a number of APIs were removed, documentation links to non existent scala/javadocs, particularly in old migration guides, had to be removed, because they would now become dead.

### CODE REVIEW

**Comment by wsargent** [comment]
Is https://github.com/playframework/playframework/issues/5677?

**Comment by jroper** [comment]
This change has resulted in binary incompatibilities that break scalatestplus-play, which breaks the play-scala template.  To deal with this, I've disabled the `play-scala` template.  We can enable it again closer to the 2.6.x release, by publishing a milestone of Play and a milestone of scalatestplus-play against that.

**Comment by jroper** [comment]
Build is green, this should be merged before more merge conflicts are introduced.

**Comment by xuwei-k** [comment]
Is there any reasons not remove [`framework/src/iteratees/src`](https://github.com/playframework/playframework/tree/5a20428d8fe23a13961371d59d03cd9a3c68e58e/framework/src/iteratees/src) ?

File: framework/src/play-streams/src/main/java/play/libs/streams/ActorFlow.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+     * @param mat The materializer to materialize the flow.
+     */
+    public static <In, Out> Flow<In, Out, ?> actorRef(Function<ActorRef, Props> props, ActorRefFactory factory, Materializer mat) {
+
+        return play.api.libs.streams.ActorFlow.<In, Out>actorRef(new AbstractFunction1<ActorRef, Props>() {
+            @Override
+            public Props apply(ActorRef v1) {
+                return props.apply(v1);
+            }
+        }, 16, OverflowStrategy.fail(), factory, mat).asJava();
```
Comment (mkurz) [inline framework/src/play-streams/src/main/java/play/libs/streams/ActorFlow.java:70]: It's interesting `fail` was used as `OverflowStrategy` here in the Java implementation of `ActorFlow`, where as of time of writing this pull request, the Scala implementation was using `dropNew`...
Related: #6246

Comment (mkurz) [reply framework/src/play-streams/src/main/java/play/libs/streams/ActorFlow.java:70]: Thanks James! Changing it to `fail` in #11529

**Pull request merged**